### PR TITLE
Implement log rotation.

### LINKED ISSUE #1417: We should store (and be able to retrieve) a "previous log"
The gateway currently stores the log for the current run, and the UI has the ability to retrieve this log.

However, if the gateway restarts itself, the cause of the restart will be the last thing in the log and will immediately be overwritten the next time the gateway starts.

I'd like to see run-app.sh modified to keep a previous log and have the UI give a way to retrieve this.

The current user study would benefit from this capability.

### PR DESCRIPTION
**Pull request by mrstegeman**
Fixes #1417

### CODE REVIEW

**Comment by mrstegeman** [comment]
@rzr This is much more cross-platform friendly and will work in cases where systemd is not available, like OpenWRT, Windows, macOS, etc.

**Comment by rzr** [comment]
OK it makes sense, so I guess if one want a "linux" profile of gateway, it should be branched and slimmed down ?

**Comment by mrstegeman** [comment]
@rzr That's an option.

**Comment by mrstegeman** [comment]
Rebased.

File: src/log-timestamps.js [hunk 1]
```diff
@@ -84,13 +111,23 @@ if (!console.constructor.hooked) {
...
   } else {
     // This path is for the normal non-jest output
-    const FUNCS = ['log', 'info', 'debug', 'error', 'warn'];
+    const FUNCS = ['info', 'debug', 'error', 'warn', 'verbose', 'silly'];
 
     for (const func of FUNCS) {
-      const realFunc = console[func];
       console[func] = function() {
-        realFunc(logPrefix() + format.apply(null, arguments));
+        const args = Array.from(arguments);
```
Comment (hobinjk) [inline src/log-timestamps.js]: This code should be replaced with `format.apply(null, arguments)` which handles objects better and is consistent with the previous code.

Comment (mrstegeman) [reply src/log-timestamps.js]: Done.

**Pull request merged**